SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+		$for2CharClass = $products.productAttributes.passwdMinimumLengthFor2CharClass
+		$for3CharClass = $products.productAttributes.passwdMinimumLengthFor3CharClass
+		$for4CharClass = $products.productAttributes.passwdMinimumLengthFor4CharClass
+		$forPhrases = $products.productAttributes.passwdMinimumCharLengthForPhrase
+		$policy = "retry=" + $numAttempts + " min=" + $for1CharClass + "," + $for2CharClass + "," + $forPhrases + "," + $for3CharClass + "," + $for4CharClass
+		Write-LogMessage -type INFO -Message "Setting ESXi hosts password policy.  Starting...."
+		Start-Sleep -s 5
+		foreach ($cid in $ppmVariables.esxiCluster.clusters.id) {
+			$cluster = Get-VCFCluster -id $cid
+			Set-EsxiPasswordPolicy -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName -cluster $cluster.name -policy $policy | Out-Null
+			Set-EsxiPasswordExpirationPeriod -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName -cluster $cluster.name -ExpirationInDays $passwdExpInDays | Out-Null
+		}
+		Write-LogMessage -type INFO -Message "Setting Password Policy on ESXi hosts.  Complete..."
+	}
+	
+	#NSX-T
+	$products = $ppmStandardConfigValues | Select-Object -Skip 0 | Where-Object -FilterScript {$_.productType -eq "NSXMgr"}
+	$apiPasswdMaxFailAttempts = $products.productAttributes.apiPasswdMaxFailAttempts
+	$apiPasswdMaxFailIntervalInSec = $products.productAttributes.apiPasswdMaxFailIntervalInSec
+	$apiPasswdUnlockIntervalInSec = $products.productAttributes.apiPasswdUnlockIntervalInSec
+	$cliPasswdMaxFailAttempts = $products.productAttributes.cliPasswdMaxFailAttempts
+	$cliPasswdMaxFailIntervalInSec = $products.productAttributes.cliPasswdMaxFailIntervalInSec
+	$minCharacterLength = $products.productAttributes.passwdMinimumLength
+	if ($ppmEnvironmentalDetails | where { ($_.productType -match "NSXMgr") -and ($_.productAttributes.driftAlarm -match "Red") } ) {
+		Write-LogMessage -type INFO -Message "Setting Password Policy on NSX-T Manager"
+		Set-NsxtManagerAuthenticationPolicy -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName -apiLockoutPeriod $apiPasswdUnlockIntervalInSec -apiResetPeriod $apiPasswdMaxFailIntervalInSec -apiMaxAttempt $apiPasswdMaxFailAttempts -cliLockoutPeriod $cliPasswdMaxFailIntervalInSec -cliMaxAttempt $cliPasswdMaxFailAttempts -minPasswdLength $minCharacterLength | Out-Null
+		Start-Sleep -s 15
+		Write-LogMessage -type INFO -Message "Setting Password Policy on NSX-T Manager.  Complete..."
+	}
+	if ($ppmEnvironmentalDetails | where { ($_.productType -match "NSXEdge") -and ($_.productAttributes.driftAlarm -match "Red") } ) {
+		Write-LogMessage -type INFO -Message "Setting Password Policy on NSX-T Edge Nodes"
```

Comment:
Change to: NSX Edge Nodes

Trying to remove NSX-T from things where we can

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e